### PR TITLE
100199 Use New Function for "Safely" Accessing the $wp_query Global

### PR DESCRIPTION
**Pull request by ggwicz**
**Ticket:** [**#100199**](http://central.tri.be/issues/100199)

**Related PRs:**
- _Tribe Common:_ https://github.com/moderntribe/tribe-common/pull/628
- _Events Calendar PRO:_ https://github.com/moderntribe/events-pro/pull/801

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [List_Widget.php:70](https://github.com/moderntribe/the-events-calendar/blob/c21da805b2ca910269a36f7002bd79c2aa2a1f22/src/Tribe/List_Widget.php#L70)

> via jenkins/codesniffer-phpcs

**Comment by mitogh** [comment]
Should this one be as `on hold` due the change on `common` module?

**Comment by mitogh** [comment]
Also I think we still need to verify if the `wp_query` object is still valid here or we shouldn't be doing the checks? 

Thanks

**Comment by ggwicz** [comment]
@mitogh, can you clarify what you mean by the following?
> Also I think we still need to verify if the wp_query object is still valid here or we shouldn't be doing the checks?

**Comment by mitogh** [comment]
Sure @ggwicz .

Basically as what's the main benefit of the new function as if someones overrides `WP_Query` or set it to null in instances like this: 

- https://github.com/moderntribe/the-events-calendar/blob/c86a5e17c2844d05fda880b858ea31b47b1d833c/src/Tribe/Integrations/X_Theme/X_Theme.php#L109

Will generate an issue so was curios if we still need to do the checks for nulls or if is an object I was under the impression we shouldn't have to do those checks.

File: src/Tribe/Bar.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public function __construct() {
 	 *
 	 */
 	public function should_show() {
-		global $wp_query;
+		$wp_query = tribe_get_global_query_object();
```
Comment (barryhughes) [inline src/Tribe/Bar.php]: Pretty much as @mitogh noted, we probably want each of these to be something like:

```php
if ( ! $wp_query = tribe_get_global_query_object() ) {
    return;
}
```

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [Main.php:844](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/Tribe/Main.php#L844)
:x: | Whitespace found at end of line | [Preview.php:106](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/Tribe/Revisions/Preview.php#L106)
:x: | Whitespace found at end of line | [Templates.php:716](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/Tribe/Templates.php#L716)
:x: | Whitespace found at end of line | [general.php:1227](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/functions/template-tags/general.php#L1227)
:x: | Whitespace found at end of line | [general.php:1605](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/functions/template-tags/general.php#L1605)
:x: | Whitespace found at end of line | [query.php:64](https://github.com/moderntribe/the-events-calendar/blob/f8c0899f58c4f535f1d1e54fa2a8769fd97d0ef4/src/functions/template-tags/query.php#L64)

> via jenkins/codesniffer-phpcs

**Comment by ggwicz** [comment]
Fixed merge conflicts and implemented @barryhughes and @mitogh's feedback. This one's ready for another review! 🔍

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [Main.php:846](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/Tribe/Main.php#L846)
:x: | Whitespace found at end of line | [Preview.php:106](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/Tribe/Revisions/Preview.php#L106)
:x: | Whitespace found at end of line | [Templates.php:716](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/Tribe/Templates.php#L716)
:x: | Whitespace found at end of line | [general.php:1244](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/functions/template-tags/general.php#L1244)
:x: | Whitespace found at end of line | [general.php:1622](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/functions/template-tags/general.php#L1622)
:x: | Whitespace found at end of line | [query.php:64](https://github.com/moderntribe/the-events-calendar/blob/49261472c0ef71ff57fd22b85071c2ad17ae562a/src/functions/template-tags/query.php#L64)

> via jenkins/codesniffer-phpcs

**Pull request merged**